### PR TITLE
Epic/pipelines/la pipelines 293 deprecated index fields filter

### PR DESCRIPTION
**Pull request by brucehyslop**
By default only return non deprecated fields when requesting list of index fields

added `deprecated` param to return deprecated fields

AtlasOfLivingAustralia/la-pipelines/issues/293

### CODE REVIEW

File: src/main/java/au/org/ala/biocache/dto/OccurrenceIndex.java [hunk 1]
```diff
@@ -92,17 +92,17 @@
...
     @Field("collectionUid")
     String collectionUid;
     @Field("collectionCode")
-    String raw_collectionCode;
+    String collectionCode;
     @Field("collectionName")
     String collectionName;
     @Field("catalogNumber")
-    String raw_catalogNumber;
+    String catalogNumber;
```
Comment (djtfmartin) [inline src/main/java/au/org/ala/biocache/dto/OccurrenceIndex.java:105]: I think this effectively changes the API (unless Im missing something) as it'll change the JSON in the basic search service. 

e.g. https://biocache.ala.org.au/ws/occurrences/search?pageSize=1

File: src/main/java/au/org/ala/biocache/dto/OccurrenceIndex.java [hunk 2]
```diff
@@ -439,12 +439,12 @@ public void setInstitutionUid(String institutionUid) {
         this.institutionUid = institutionUid;
     }
 
-    public String getRaw_institutionCode() {
-        return raw_institutionCode;
+    public String getInstitutionCode() {
```
Comment (djtfmartin) [inline src/main/java/au/org/ala/biocache/dto/OccurrenceIndex.java:442]: Ditto the above

**Review by djtfmartin — COMMENTED** [review]
I think the changes to OccurrenceIndex (which would improve things no doubt) will change the existing API and cause biocache UI to break.

File: src/main/java/au/org/ala/biocache/dto/OccurrenceIndex.java (hunk 2 above)
Comment (brucehyslop) [reply src/main/java/au/org/ala/biocache/dto/OccurrenceIndex.java:442]: It would also affect any consumers of the search service.

Comment (brucehyslop) [reply src/main/java/au/org/ala/biocache/dto/OccurrenceIndex.java]: Change made in commit https://github.com/AtlasOfLivingAustralia/biocache-service/pull/598/commits/9801972bcf1841ee299ea642629e4d663bb8f9b4

**Pull request merged**